Server/VikingVault.API/Controllers/TransferRequestsController.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using VikingVault.DataAccess.Models;
+using VikingVault.DataAccess.Models.Exceptions;
+using VikingVault.Services.Abstractions;
+using VikingVault.Services.Exceptions;
+using VikingVault.Services.Exceptions.CardException;
+
+namespace VikingVault.API.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class TransferRequestsController : ControllerBase
+    {
+        ITransferRequestService _transferRequestService;
+        IUserService _userService;
+
+        public TransferRequestsController(ITransferRequestService transferRequestService, IUserService userService)
+        {
+            _transferRequestService = transferRequestService;
+            _userService = userService;
+        }
+
+        [HttpPost]
+        public ActionResult<string> Post([FromBody] TransferRequestDTO transferRequestDTO)
+        {
+            try
+            {
+                var token = Request.Headers["x-access-token"];
+
+                if(transferRequestDTO != null)
+                {
+                    var requester = _userService.GetUserFromToken(token);
+                    var reciever = _userService.GetUserFromCardNumber(transferRequestDTO.CardNumberReciever);
+
+                    if(requester.Id != reciever.Id)
+                    {
+                        var transferRequestData = transferRequestDTO.ConvertDTOtoTransferRequest(requester);
+                        _transferRequestService.AddTransferRequest(transferRequestData);
+
+                        return Ok("Succesfully requested " + transferRequestData.Amount + "!");
+                    }
+                    else
+                    {
+                        return Ok("Select a different user to request money from!");
+                    }                   
+                }
+                else
+                {
+                    return NotFound("Request to server unsuccesful.");
+                }
+            }
+            catch (Exception e)
+            {
+                if (e is NoCardAttachedToUserException || e is TransactionException || e is DatabaseException)
+                {
+                    return NotFound(e.Message);
+                }
+
+                return NotFound("Unknown error.");
```

Comment:
Usually if something goes wrong but you don't know what's wrong it's an 500 Internal Server Error, a NotFound has the status code of 404 which simply menans that the resource that you've asked couldn't be found, but everything else worked as expected. But in here the NotFound is thrown at the end of the `catch( Exception)`

Reply:
Fair enough. Changed it